pkg/asset/agent/manifests/clusterdeployment.go

```diff
@@ -39,21 +42,77 @@ func (*ClusterDeployment) Dependencies() []asset.Asset {
 }
 
 // Generate generates the ClusterDeployment manifest.
-func (i *ClusterDeployment) Generate(dependencies asset.Parents) error {
-	// TODO: generate from install-config
-	return i.finish()
+func (cd *ClusterDeployment) Generate(dependencies asset.Parents) error {
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
+
+	if installConfig.Config != nil {
+		clusterDeployment := &hivev1.ClusterDeployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ClusterDeployment",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      installConfig.Config.ObjectMeta.Name,
+				Namespace: installConfig.Config.ObjectMeta.Namespace,
+			},
+			Spec: hivev1.ClusterDeploymentSpec{
+				ClusterName: installConfig.Config.ObjectMeta.Name,
+				BaseDomain:  installConfig.Config.BaseDomain,
+				Platform: hivev1.Platform{
+					AgentBareMetal: &hivev1Agent.BareMetalPlatform{
+						AgentSelector: metav1.LabelSelector{
+							// What labels should be set here? Will the values come from install config?
+							MatchLabels: map[string]string{},
+						},
+					},
+				},
+				// What should be set as PullSecretRef name here? Will the values come from install config?
+				PullSecretRef: &v1.LocalObjectReference{Name: "pull-secret"},
+				ControlPlaneConfig: hivev1.ControlPlaneConfigSpec{
+					ServingCertificates: hivev1.ControlPlaneServingCertificateSpec{
+						// Default: "",
+						// Additional: []hivev1.ControlPlaneAdditionalCertificate{{
+						// 	Name:   clusterDeploymentFilename,
+						// 	Domain: "",
+						// }},
+					},
```

Comment:
@zaneb :point_up:  From where will we read `ControlPlaneAdditionalCertificate` data?

Reply:
Fixed.